### DISCUSSION TITLE
Webserver

### DISCUSSION THREAD

**Discussion by DityOldDogDK** [PlummersSoftwareLLC/NightDriverStrip#169] [Q&A]
Hi
I am trying to make the webserver on my DOIT ESP32 DEVKIT V1,but i dont know how to do it i have been reading the setup but i dont know how to.

Webserver Setup
To use the built-in webserver, you will need to build and upload the SPIFFS image to your board's flash using platformio.
You can do this using the platformio user interface, or using the pio command line tool

pio run --target buildfs --environment <project name>
pio run --target uploadfs --environment <project name>

**Comment by erikpt** [comment]
Here's the short version based on what I've been able to do: 

Open the NightDriver project in Visual Studio Code

Open the file: partitions_custom.csv
Make the edits as specified in this post: https://github.com/PlummersSoftwareLLC/NightDriverStrip/discussions/159#discussioncomment-4247605

On the bottom of the window you should see a bar with icons similar to the following:
![image](https://user-images.githubusercontent.com/4423348/205462510-490a8465-f2cc-4200-8803-aff569074a30.png)

Click this icon (circled in blue) to open the terminal: 
![image](https://user-images.githubusercontent.com/4423348/205462614-977a7625-75ee-449f-a037-e9a06edce412.png)

Connect your board to the computer via USB cable.

Run the following commands in order: 
pio run
pio run --target upload
pio run --target buildfs --environment
pio run --target uploadfs --environment

**Reply by DityOldDogDK** [reply]
I get a error when i get to pio run --target buildfs --environment

SPIFFS_write error(-10001): File system is full.

error adding file!
Error for adding content from js!

**Reply by erikpt** [reply]
DId you edit all three lines in the partitions_custom.csv so they match this:

nvs, data, nvs, 0x009000, 0x005000,
otadata, data, ota, 0x00e000, 0x002000,
app0, app, ota_0, **0x010000**, **0x150000**,
app1, app, ota_1, **0x160000**, **0x150000**,
storage, data, spiffs, **0x2B0000**, **0x150000**

I've bolded all the edits you need to make.  If you're using a board that has 4GB flash then that's a requirement./

**Reply by CdnLaser** [reply]
Save error as the OP. Confirmed the changes....
/static/css/main.0f983434.css
/static/js/main.js.gz
SPIFFS_write error(-10001): File system is full.

error adding file!
Error for adding content from js!

**Reply by erikpt** [reply]
In your platformio.ini file search for "build_type" item.  This may either be in the main [env] section or on your board's definition.  If either is set to debug, try changing them both to release, as seen below.

```
[env]
platform        = espressif32
framework       = arduino
**build_type      = release**
build_unflags   = -std=gnu++11
```

Others here with more knowledge than I have mentioned this may make the builds smaller. Not sure that applies to SPIFFS filesystem as well but worth a shot.  Also, can you post your partitions_custom.csv file?

**Reply by CdnLaser** [reply]
Added build_type  =  release
seemed to buildfs and uploadfs appeared to have worked this time.
Dont understand why some of the 'projects' have 'build_type = debug' some have 'build_type = release' and some have none. 
I'm a hardware guy so 'trying' to learn this is frustrating when its not documented and incomplete :\ oh well will keep plugging away
Here is my current  partitions_custom.cvs

Still cant seem to get the web server to come up on my 192.168.0.35:80 
Will keep at it..

# ESP-IDF Partition Table
# This gives us some additional space for code.
# It should also fix the OTA regression.
# Name,   Type, SubType,     Offset,      Size, Flags

nvs,        data,   nvs,     0x009000,    0x005000,
otadata,    data,   ota,     0x00e000,    0x002000,
app0,       app,    ota_0,   0x010000,    0x150000,
app1,       app,    ota_1,   0x160000,    0x150000,
storage,    data,   spiffs,  0x2B0000,    0x150000

**Comment by davepl** [comment]
Here’s my partition file, which is not checked in, but should allows you build and upload everything… works for me with my build, anyway!

# ESP-IDF Partition Table
# This gives us some additional space for code. 
# It should also fix the OTA regression.  
# Name,   Type, SubType,     Offset,      Size, Flags

nvs,        data,   nvs,     0x009000,    0x005000,
otadata,    data,   ota,     0x00e000,    0x002000,
app0,       app,    ota_0,   0x010000,    0x150000,
app1,       app,    ota_1,   0x160000,    0x150000,
storage,    data,   spiffs,  0x2B0000,    0x150000

Note that I’ve only tested the web server under the mesmerizer config, which has more base ram available because it’s using PSRAM.

We may have to offload or move some of the React templates to the web, rather than hosting everything off the chip, but that’d need someone who’s better with React than I am!

-Dave

**Comment by davepl** [comment]
In a very recent checkin I removed the built_type from individual projects, so its now global and the same for all… and ideally retail!
